tests/test_notify_v2.py

```diff
@@ -0,0 +1,415 @@
+        self.assertEquals(notification['match']['bundle_uuid'], bundle_uuid)
+        self.assertEquals(notification['match']['bundle_version'], f"{bundle_version}.dead")
+
+    @testmode.integration
+    def test_unversioned_tombstone_notifications(self, replica=Replica.aws):
+        bucket = get_env('DSS_S3_BUCKET_TEST')
+        notification_object_key = f"notification-v2/{uuid4()}"
+        url = self.s3.generate_presigned_url(
+            ClientMethod='put_object',
+            Params=dict(Bucket=bucket, Key=notification_object_key, ContentType="application/json")
+        )
+        subscription = self._put_subscription(
+            {
+                'callback_url': url,
+                'method': "PUT",
+                'jmespath_query': "admin_deleted==`true`"
+            },
+            replica
+        )
+        bundle_uuid, bundle_version = self._upload_bundle(replica)
+        bundle_uuid, bundle_version = self._upload_bundle(replica, bundle_uuid)
+        self._tombstone_bundle(replica, bundle_uuid)
+
+        notification = self._get_notification_from_s3_object(bucket, notification_object_key)
+        self.assertEquals(notification['subscription_id'], subscription['uuid'])
+        self.assertEquals(notification['match']['bundle_uuid'], bundle_uuid)
+        # TODO:
+        # Multiple notifications should be delivered for this test. However, the notification
+        # test infrastructure (presigned s3 url) cannot track multiple deliveries. Need another
+        # mechanism.
+        # Brian Hannafious 2019-01-30
+
+    @testmode.integration
+    def test_queue_notification(self):
+        replica = Replica.aws
+        bucket = get_env('DSS_S3_BUCKET_TEST')
+        key = f"notification-v2/{uuid4()}"
+        post = self.s3.generate_presigned_post(
+            Bucket=bucket,
+            Key=key,
+            ExpiresIn=60,
+            Fields={'Content-Type': "application/json"},
+            Conditions=[{'Content-Type': "application/json"}]
+        )
+        subscription = self._put_subscription(
+            {
+                'payload_form_field': "file",
+                'form_fields': post['fields'],
+                'callback_url': post['url'],
+                'encoding': "multipart/form-data",
+            },
+            replica
+        )
+
+        queue_notification(
+            replica,
+            subscription,
+            "CREATE",
+            "bundles/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z",
+            delay_seconds=0
+        )
+        notification = self._get_notification_from_s3_object(bucket, key)
+        self.assertEquals(notification['subscription_id'], subscription['uuid'])
+
+    @testmode.integration
+    def test_bundle_notification(self):
+        for replica in Replica:
+            with self.subTest(replica):
+                self._test_bundle_notification(replica)
+
+    def _test_bundle_notification(self, replica):
```

Comment:
do you have a test where you upload different versions of the same bundle?

Reply:
No but I'm not sure I see the point since that's just another creation event from the buckets.